components/Base/Button.vue

```diff
@@ -0,0 +1,95 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import type { RouteLocationRaw } from 'vue-router';
+import ArrowLeft from '~/assets/icons/arrow-left.svg?skipsvgo';
+import ArrowRight from '~/assets/icons/arrow-right.svg?skipsvgo';
+
+const props = withDefaults(
+  defineProps<{
+    variant?: 'fill' | 'outline';
+    type?: 'regular' | 'round' | 'sqaure';
+    to?: RouteLocationRaw | null;
+    href?: string | null;
+    arrow?: null | 'right' | 'left';
+  }>(),
+  {
+    variant: 'fill',
+    type: 'regular',
+    to: null,
+    href: null,
+    arrow: null,
+  },
+);
+
+const componentType = computed(() => {
+  if (props.href) return 'a';
+  if (props.to) return 'NuxtLink';
+  return 'button';
+});
+
+const buttonClasses = computed(() => {
+  if (componentType.value === 'a' || componentType.value === 'NuxtLink')
```

Comment:
How does it work? is there no variants and types for links? 🤔

Reply:
```
const componentType = computed(() => {
  if (props.href) return 'a';
  if (props.to) return 'NuxtLink';
  return 'button';
});
```

Reply:
I love this dynamic component base-buttons and I personally implement it everywhere 😄

Reply:
yeah, this part with choosing component is cool but here we have styling of components and it looks like variants and types are applied only for buttons 🤔 I think we should make possibility to add these also for links cause a lot of buttons in the app will be links 👌